### DISCUSSION TITLE
Model WishList

### DISCUSSION THREAD

**Discussion by saddam213** [saddam213/OnnxStack#66] [General]
As you know I have no idea or way to convert models to ONNX format, I rely solely on the community

There are a few models I need in ONNX format to help in my development efforts, if anyone out there would like to convert some of there for me it would be epic!!!

## StableDiffusion
- [ ] PixArt-LCM-XL  - [PixArt-LCM-XL-2-1024-MS](https://huggingface.co/PixArt-alpha/PixArt-LCM-XL-2-1024-MS)
- [x] Instruct-Pix2Pix - [ForserX/instruct-pix2pix-onnx](https://huggingface.co/ForserX/instruct-pix2pix-onnx) **Done!**

## AnimateDiff 
- [ ] AnimateDiff - [animatediff](https://huggingface.co/guoyww/animatediff/tree/main)
- [ ] AnimateDiff_ControlNet [animatediff_controlnet](https://huggingface.co/crishhh/animatediff_controlnet/tree/main)

## ControlNet
- [x] Control_SD15_Canny - [ForserX/sd-controlnet-canny-onnx](https://huggingface.co/ForserX/sd-controlnet-canny-onnx) **Done!**
- [x] Control_SD15_Depth - [ForserX/sd-controlnet-depth-onnx](https://huggingface.co/ForserX/sd-controlnet-depth-onnx) **Done!**
- [x] Control_SD15_HED - [ForserX/sd-controlnet-hed-onnx](https://huggingface.co/ForserX/sd-controlnet-hed-onnx) **Done!**
- [x] Control_SD15_MLSD - [ForserX/sd-controlnet-mlsd-onnx](https://huggingface.co/ForserX/sd-controlnet-mlsd-onnx) **Done!**
- [x] Control_SD15_Normal - [ForserX/sd-controlnet-normal-onnx](https://huggingface.co/ForserX/sd-controlnet-normal-onnx) **Done!**
- [x] Control_SD15_OpenPose - [ForserX/sd-controlnet-openpose-onnx](https://huggingface.co/ForserX/sd-controlnet-openpose-onnx) **Done!**
- [x] Control_SD15_Scribble - [ForserX/sd-controlnet-scribble-onnx](https://huggingface.co/ForserX/sd-controlnet-scribble-onnx) **Done!**

## Upscalers:
- [x] 003_realSR_BSRGAN_DFO_s64w8_SwinIR-M_x4_GAN - [uwg/upscaler](https://huggingface.co/uwg/upscaler/tree/main/SwinIR) **Done!**
- [ ] 003_realSR_BSRGAN_DFO_s64w8_SwinIR-M_x2_GAN - [uwg/upscaler](https://huggingface.co/uwg/upscaler/tree/main/SwinIR)
- [ ] 001_classicalSR_DF2K_s64w8_SwinIR-M_x2 - [uwg/upscaler](https://huggingface.co/uwg/upscaler/tree/main/SwinIR)
- [ ] 001_classicalSR_DF2K_s64w8_SwinIR-M_x3 - [uwg/upscaler](https://huggingface.co/uwg/upscaler/tree/main/SwinIR)
- [ ] 001_classicalSR_DF2K_s64w8_SwinIR-M_x4 - [uwg/upscaler](https://huggingface.co/uwg/upscaler/tree/main/SwinIR)
- [ ] 001_classicalSR_DF2K_s64w8_SwinIR-M_x8 - [uwg/upscaler](https://huggingface.co/uwg/upscaler/tree/main/SwinIR)

## TAESD
- [x] TAESD - [taesd-onnx](https://huggingface.co/deinferno/taesd-onnx) - **Done!**
- [ ] TAESD-XL [taesdxl](https://huggingface.co/madebyollin/taesdxl)

## LoRA
- [x] LCM-LoRA-SD - [Disty0/LCM_SoteMix](https://huggingface.co/Disty0/LCM_SoteMix)
- [ ] LCM-LoRA-XL - [lcm-lora-sdxl](https://huggingface.co/latent-consistency/lcm-lora-sdxl)

**Comment by Amin456789** [comment]
forserx's page has onnx pix2pix and controlnet =
https://huggingface.co/ForserX

@ashd @forserx can u guys please help our friend for converting these models?

btw, Stable Video Diffusion img2vid is a very good img to video, maybe easier than animatediff to impelement

**Reply by saddam213** [reply]
Thanks for that link, thats heaps off the list :)

**Reply by AshD** [reply]
Does this model cover the need for LCM-Lora? 
https://huggingface.co/softwareweaver/Latent-Consistency-xl-Olive-Onnx

**Reply by ForserX** [reply]
I did video generation as follows: I split the input video into 24 frames per second, processed a packet of frames, after which I baked back the video from the received images via ffmpeg/opencv.

**Comment by ForserX** [comment]
For upscalers, I'll say this: there are C++ implementations with vulkan models. They work out less than a second. If you are interested, I can give you links.

**Comment by ForserX** [comment]
https://huggingface.co/softwareweaver/pixel-art-xl-Olive-Onnx/tree/main

**Comment by AshD** [comment]
I added a SDXL DPO model
https://huggingface.co/softwareweaver/sdxl-dpo-Olive-Onnx

Direct Preference Optimization (DPO) for text-to-image diffusion models is a method to align diffusion models to text human preferences by directly optimizing on human comparison data. https://arxiv.org/abs/2311.12908

In my limited testing, It seems to follow my prompts better.

**Reply by saddam213** [reply]
Nice, downloading now...

**Reply by AshD** [reply]
Added a new model that combines DPO with LCM. Should have the features of DPO with the speed of LCM.

https://huggingface.co/softwareweaver/SDXL-DPO-LCM-Olive-Onnx

**Comment by saddam213** [comment]
Anyone had any luck converting Animatediff or StableDiffucionVideo models?

would be awesome to add support for those

**Comment by saddam213** [comment]
LCM-LoRA-SD - [Disty0/LCM_SoteMix](https://huggingface.co/Disty0/LCM_SoteMix)
